app/main.py

```diff
@@ -2,0 +1,33 @@
+from __future__ import annotations
+
+
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(obj: "Herbivore") -> None:
```

Comment:
The `bite` method assumes that the `obj` parameter is always a `Herbivore`. While the `isinstance` check on line 29 ensures this, it might be more robust to handle the case where `obj` is not a `Herbivore` more explicitly, perhaps by raising an exception or returning early.